### ISSUE TITLE
marking of a rudimentary description ("stub")

### ISSUE THREAD

**Issue by DenisNosnitsin1970** [SChAth/ScAthiop#2700]
One question (possibly I overlooked this): is it possible to mark a rudimentary description as such (if the cataloguer wants others consider it as "stub")?
All description have the same indication "work in progress", but "stub" is more than that.

**Comment by eu-genia** [comment]
I put the word `stub `in `change`

**Comment by eu-genia** [comment]
at the moment if you have no `change `element (or the only `change `element has `who="PL"` then "Stub" is printed 
![image](https://github.com/user-attachments/assets/b0d4d049-51f1-4caa-803c-993f66bc49b3)


```
   {if (count($document//t:change[not(@who eq 'PL')]) eq 1) then
   <span class="w3-tag w3-red">Stub</span>
   else if ($document//t:change[contains(.,'completed')]) then
   <span class="w3-tag w3-gray" >Under Review</span>
     else if ($document//t:change[contains(.,'reviewed')]) then
   <span class="w3-tag w3-white" >Version of {max($document//t:change/xs:date(@when))}</span>
   else
<span class="w3-tag w3-red" >{"Work in Progress"}</span>
```

we can agree on when "Stub" must be printed

**Comment by eu-genia** [comment]
I think it is not present in the guidelines, we should agree on the list of values and visalization and add it

**Comment by eu-genia** [comment]
@nafisa-valieva @CarstenHoffmannMarburg @karljonaskarlsson @thea-m @abausi 
I think this is another small but important addition to the guidelines that we should all agree upon and implement: how we mark stub records and whether and how we want them visualized.

First I thought that the easiest would be that in all cases where only 1 `change `element is present AND this element contains the word "**stub**" we see the word "**Stub**" (as in the screenshot above). Then, as soon as there is a second `change `element, or there is just one `change `element but it does not contain the term **stub** this term would not show. 

This sounds OK but we can also have stubs where changes occurred (like `facs `added, or something else) making a second `change `element necessary but the record is still a stub. 

We could possibly say that as long as there is the word **stub** anywhere in the `change `history then "Stub" is printed, and when we think the record has stopped being a stub as it has been developed further then in the first change line the word **stub** must replaced with something like **initial version** or anything else. It is a bit awkward to have to modify earlier `change `elements but at the moment this is something that I can see as the easiest to implement. Other ideas?

**Comment by nafisa-valieva** [comment]
> @nafisa-valieva @CarstenHoffmannMarburg @karljonaskarlsson @thea-m @abausi I think this is another small but important addition to the guidelines that we should all agree upon and implement: how we mark stub records and whether and how we want them visualized.
> 
> First I thought that the easiest would be that in all cases where only 1 `change `element is present AND this element contains the word "**stub**" we see the word "**Stub**" (as in the screenshot above). Then, as soon as there is a second `change `element, or there is just one `change `element but it does not contain the term **stub** this term would not show.
> 
> This sounds OK but we can also have stubs where changes occurred (like `facs `added, or something else) making a second `change `element necessary but the record is still a stub.
> 
> We could possibly say that as long as there is the word **stub** anywhere in the `change `history then "Stub" is printed, and when we think the record has stopped being a stub as it has been developed further then in the first change line the word **stub** must replaced with something like **initial version** or anything else. It is a bit awkward to have to modify earlier `change `elements but at the moment this is something that I can see as the easiest to implement. Other ideas?

This looks good for me! I am only afraid that we will never delete the word "stub" at the end :-)

**Comment by eu-genia** [comment]
Yes I am also afraid but since I always ask you to check how the record you have created/merged appears online would one not notice that the red "Stub" is there when it should not be?
I am really not sure that that is the best way, revising historical `change ` is not the best practice, I am just trying to think of a way to somehow clearly distinguish stubs at first glance, without looking into the XML.

**Comment by DenisNosnitsin1970** [comment]
This sounds best for me: "We could possibly say that as long as there is the word stub anywhere in the change history then "Stub" is printed, and when we think the record has stopped being a stub as it has been developed further then in the first change line the word stub must replaced with something like initial version or anything else". The cataloguer has to remove the stub-marking as soon as he/she does not consider the description a stub any longer (if I understood it right). Yes, the stubs should be distinguishable at first, glance.

**Comment by eu-genia** [comment]
In this case we would gave to check first all existing records that use `stub` in `change` and replace it when needed
https://betamasaheft.eu/xpath.html?xpath=%24config%3Acollection-root%2F%2Ft%3ArevisionDesc%2Ft%3Achange%5Bcontains%28.%2C%27tub%27%29%5D

**Comment by CarstenHoffmannMarburg** [comment]
I agree with Eugenia initial suggestion. It should be a solution, that is easy to handle and does not require awareness or action on the side of cataloguers, who will surely change in the future as well. I think, it is not a problem, if some rudimentary descriptions remain, that are not marked as "stubs" explicitely.

**Comment by karljonaskarlsson** [comment]
Eugenia's suggestion makes sense to me as well!

**Comment by thea-m** [comment]
Do the guidelines also need to be updated regarding this? https://betamasaheft.eu/Guidelines/?id=revisions

**Comment by eu-genia** [comment]
Yes, thank you!

**Comment by nafisa-valieva** [comment]
> Do the guidelines also need to be updated regarding this? https://betamasaheft.eu/Guidelines/?id=revisions

Probably it makes sense to work on both updates together. If you do not mind, I will take care of it and then ask for your revision.

**Comment by thea-m** [comment]
Thank you @nafisa-valieva that's great!

**Comment by nafisa-valieva** [comment]
Dear @DenisNosnitsin1970, according to your wishes Eugenia changed the encoding and I have already introduced this discussion into the guidelines:
see [Differentiating](https://betamasaheft.eu/Guidelines/?id=manuscripts) Between Various Levels of Description

so I think you can close this issue!

**Comment by eu-genia** [comment]
Sorry let us keep it open for a bit as I will need to update the xquery to highlight the stubs properly but I still have to wait for all app modifications till the new version is there... should wait

**Comment by nafisa-valieva** [comment]
Dear @eu-genia, I do not know whether it is relevant or you know it already, but I have noticed that many records that do not have a word "stub" in the change history and are rather complete, are marked as "stub" now.

**Comment by eu-genia** [comment]
I have not yet changed anything; there is still an old rule for stubs in place. Opening an issue and adding something to guidelines is not sufficient to see anything in the application.

Changing the visualization is not only tricky - since it applies everywhere, also to the past records, one has to be sure to not introduce more errors than one resolves. 

You can also check the message I had left above yours in this issue.

**Comment by nafisa-valieva** [comment]
> I have not yet changed anything; there is still an old rule for stubs in place. Opening an issue and adding something to guidelines is not sufficient to see anything in the application.
> 
> Changing the visualization is not only tricky - since it applies everywhere, also to the past records, one has to be sure to not introduce more errors than one resolves.
> 
> You can also check the message I had left above yours in this issue.

Yes, I read it! but then I do not understand how the old rule worked. You wrote `at the moment if you have no change element (or the only change element has who="PL"`, but BNF records do not have who=PL.

**Comment by nafisa-valieva** [comment]
> > I have not yet changed anything; there is still an old rule for stubs in place. Opening an issue and adding something to guidelines is not sufficient to see anything in the application.
> > Changing the visualization is not only tricky - since it applies everywhere, also to the past records, one has to be sure to not introduce more errors than one resolves.
> > You can also check the message I had left above yours in this issue.
> 
> Yes, I read it! but then I do not understand how the old rule worked. You wrote `at the moment if you have no change element (or the only change element has who="PL"`, but BNF records do not have who=PL.
I cannot be sure for 100%, but my impression is that files that were not marked as stubs became marked as such. That's why I signalled it to you.

**Comment by eu-genia** [comment]
The rule is posted above

   ```
if (count($document//t:change[not(@who eq 'PL')]) eq 1) then
   <span class="w3-tag w3-red">Stub</span>
   else if ($document//t:change[contains(.,'completed')]) then
   <span class="w3-tag w3-gray" >Under Review</span>
     else if ($document//t:change[contains(.,'reviewed')]) then
   <span class="w3-tag w3-white" >Version of {max($document//t:change/xs:date(@when))}</span>
   else
<span class="w3-tag w3-red" >{"Work in Progress"}</span>
```

If there is just one change element then stub
if there is a change element with completed then Under review
if there is a change element with reviewed then Version...
else Work in Progress

**Comment by eu-genia** [comment]
(it was enough to add a second change element to make "stub" disappear. the idea was that records are reviewed and therefore no record may have just one change element unless it is a stub)

**Comment by nafisa-valieva** [comment]
> The rule is posted above
> 
> ```
> if (count($document//t:change[not(@who eq 'PL')]) eq 1) then
> <span class="w3-tag w3-red">Stub</span>
> else if ($document//t:change[contains(.,'completed')]) then
> <span class="w3-tag w3-gray" >Under Review</span>
>   else if ($document//t:change[contains(.,'reviewed')]) then
> <span class="w3-tag w3-white" >Version of {max($document//t:change/xs:date(@when))}</span>
> else
> <span class="w3-tag w3-red" >{"Work in Progress"}</span>
> ```
> 
> If there is just one change element then stub if there is a change element with completed then Under review if there is a change element with reviewed then Version... else Work in Progress

Thank you for this explanation!

**Comment by eu-genia** [comment]
I have been trying to see how the new rule can be implemented but we run the danger that many stubs that have no "stub" in the change element as they were created according to the old rule will seem more advanced than they are.

I suggest that we have the double rule
- "stub" is printed if the word appears in the `change `element AND
- if there is only one `change `element

an advanced record should never have just one change element as after the record had been created there must have been some review and some changes involved. each commit should ideally be associated with a new change element (unless it is a really meaningless correction)

so if you have records that you do not want to have the word "stub" shown please check if an additional `change `element should be added

**Comment by CarstenHoffmannMarburg** [comment]
> an advanced record should never have just one change element as after the record had been created there must have been some review and some changes involved. each commit should ideally be associated with a new change element (unless it is a really meaningless correction)

Each of the records for BL and BML manuscripts, which I have created on the basis of manuscript descriptions do only have one change element, but I would regard them as rather elaborated in comparison to many others. I would suggest to call a stub only if, this is stated in the change element. It is better to have many short records without the label "stub" rather than the opposite with many elaborated records, that are wrongly attributed "stubs".

**Comment by eu-genia** [comment]
The rule was that each commit (after review you usually do revisions) should be associated with a change element describing what you did. So rather that the elaborate records should not have one change elements and should be corrected.

The description should be in `change `(and not in github subject line)

Or we will have thousands of EMIP and EMML records not appearing as stubs any more

**Comment by eu-genia** [comment]
I can remove the `one change = stub` rule and we can see whether we are happier with the result

**Comment by eu-genia** [comment]
I have removed the rule but of course now we have no distinction between a record like 
BLorient13278 (one change element) and
BLorient15421 (one change element)

I still think that the best and clearest way to have more change elements for developed records (or someone must introduce the word stub to all the stubs, and there are many more stubs than developed records)

**Comment by DenisNosnitsin1970** [comment]
Yes, indeed, now there is no easy quick and formal way to distinguish between them. Perhaps introducing "stub" may be a bit time consuming, but the results will be clear, and it will be easy to implement it in the future (I can include "stub" into all collections I introduced within the last couple  of years).

**Comment by eu-genia** [comment]
I think we need to introduce "stub" as a term in particular because we might have also for stubs several `change `elements, which used to be "invisible".
At the same time developed records may not have just one `change` element as then the entire peer review process becomes invisible

**Comment by DenisNosnitsin1970** [comment]
I commonly had in stubs more than one element, usually three.

**Comment by eu-genia** [comment]
Exactly. This is now overcome by the script, one word stub is enough and it will be displayed as stub.
But there should not be developed records with just one change element.

Unfortunately what had been discussed and implented in the application code never made it to the guidelines (or rather it was there in the wiki but something has moved so cannot be found any more, just some traces under Discussions https://github.com/BetaMasaheft/Documentation/discussions/1855) so we all forgot what we had decided:

The rule is (used to be):

**1) just _one_ `change `element -> show as "stub". add more change elements when reviewing and updating so that this disappears immediately once there has been work done. ((now the word "stub" is shown whenever it appears in any of the change elements.))**

**2) introduce the word "reviewed" in the change element to have the "Version of XX" printed**

**3) introduce the word "completed" in the change element to have "Under Review" printed**

**4) anything else is shown as "Work in Progress"**


Do we need to change this entire set of rules? Do we need them? Surely this must be added to the Guidelines or noone will ever remember.

**Comment by DenisNosnitsin1970** [comment]
"This is now overcome by the script, one word stub is enough and it will be displayed as stub." - I agree with that.

**Comment by eu-genia** [comment]
To sum up, **the rule for Stubs** should be:

- if you WANT your record to be seen as "**stub**" - put the word "stub" in the `change `element
- if you DO NOT want your record to be seen as "stub" - make sure there is **more than one** `change `element, or it will be still shown as "stub" (I will put this back in the xquery as it does not seem right to me as it is now, it is more correct to update elaborate records with one change element to respect this rule) **and** make sure that there is **no word "stub"** in any of the change elements (revise if necessary)


What about the other rules?
- if you want to show that the record is as complete as it can be as far as you are concerned you can put the word "completed" in the `change `element, in this case the words "Under Review" will be printed (needed? other words?)
- if you want to show that the record has undergone a review process you can put the word "reviewed" in the change element and then you will see "Version of (date of review)" printed
- all others are "Work in progress"

**Comment by DenisNosnitsin1970** [comment]
ok

**Comment by eu-genia** [comment]
Another thing I could do is that when a record is created through the app, the change element generated is always "created stub" (not entity as now) . Then everyone will remember to modify/add (?)

**Comment by karljonaskarlsson** [comment]
> Another thing I could do is that when a record is created through the app, the change element generated is always "created stub" (not entity as now) . Then everyone will remember to modify/add (?)

Or we will remember to remind each other in the reviewing process

**Comment by eu-genia** [comment]
manuscripts by CH  with one change element: https://betamasaheft.eu/xpath.html?xpath=%24config%3Acollection-rootMS%2F%2Ft%3ArevisionDesc%5Bcount%28t%3Achange%29+eq+1%5D%2Ft%3Achange%5Bcontains%28%40who%2C%27CH%27%29%5D
works:
https://betamasaheft.eu/xpath.html?xpath=%24config%3Acollection-rootW%2F%2Ft%3ArevisionDesc%5Bcount%28t%3Achange%29+eq+1%5D%2Ft%3Achange%5Bcontains%28%40who%2C%27CH%27%29%5D

manuscripts by JK with one change element:
https://betamasaheft.eu/xpath.html?xpath=%24config%3Acollection-rootMS%2F%2Ft%3ArevisionDesc%5Bcount%28t%3Achange%29+eq+1%5D%2Ft%3Achange%5Bcontains%28%40who%2C%27JK%27%29%5D
works:
https://betamasaheft.eu/xpath.html?xpath=%24config%3Acollection-rootW%2F%2Ft%3ArevisionDesc%5Bcount%28t%3Achange%29+eq+1%5D%2Ft%3Achange%5Bcontains%28%40who%2C%27JK%27%29%5D

etc

**Comment by CarstenHoffmannMarburg** [comment]
> manuscripts by CH with one change element: https://betamasaheft.eu/xpath.html?xpath=%24config%3Acollection-rootMS%2F%2Ft%3ArevisionDesc%5Bcount%28t%3Achange%29+eq+1%5D%2Ft%3Achange%5Bcontains%28%40who%2C%27CH%27%29%5D
> works:
> https://betamasaheft.eu/xpath.html?xpath=%24config%3Acollection-rootW%2F%2Ft%3ArevisionDesc%5Bcount%28t%3Achange%29+eq+1%5D%2Ft%3Achange%5Bcontains%28%40who%2C%27CH%27%29%5D

As far as I can remember, I made most of the records for works, manuscripts, persons, places, authFile with only one change element. If I understand you correctly, you want me to check all these records and add a change element to each one.

**Comment by CarstenHoffmannMarburg** [comment]
> The rule is (used to be):
> 
> **1) just _one_ `change `element -> show as "stub". add more change elements when reviewing and updating so that this disappears immediately once there has been work done. ((now the word "stub" is shown whenever it appears in any of the change elements.))**
> 
> **2) introduce the word "reviewed" in the change element to have the "Version of XX" printed**
> 
> **3) introduce the word "completed" in the change element to have "Under Review" printed**
> 
> **4) anything else is shown as "Work in Progress"**
> 
> Do we need to change this entire set of rules? Do we need them? Surely this must be added to the Guidelines or noone will ever remember.

Personally, I do not see any benefits from the distinction of 2), 3) and 4). Infact these attributes are more or less misleading and mostly given arbitrarily, i.e.:
2) Seeing "Version of XX", many users may think, that there is another version of this record, what is not the case except for outdated previous versions.
3) Seeing "Under review", many users may think, that the record is currently in a review process, what is not the case.
4) Seeomg "Work in Progress" some users will think, that somebody is currently working on this file, while most users may think, that this is just a general statement that applies to every record.

In the end, I would recommend to abstain from these attributes except for 1) "stub", because this is the only relevant one in my perspective.

**Comment by eu-genia** [comment]
In order to save us from too much work for the moment I have introduced

word "stub" = "stub" appears
one change element = "draft" appears, instead of "stub". this indicates that more `change `elements may be needed. (i would not add it to the guidelines, this is a temporary solution)

as for more `change `elements - we can have "**work in progress**" for all or also "**version of XXXX-XX-XX**" - exactly, this implies that this is the version created on day XX and that **there are earlier versions**, this is necessary if someone quotes the link in an article, as then one can quote exactly the version as it was on that day, even if it may change in the future

indeed I am also not convinced that reviewed/completed are sensible and it is all very difficult to keep in mind

this all was born from an old discussion that we do not want records to be considered "final" in the same way printed articles are perceived - unless they are marked by the creator as "completed" (which is of course always subjective)

**Comment by CarstenHoffmannMarburg** [comment]
> this implies that this is the version created on day XX and that **there are earlier versions**, this is necessary if someone quotes the link in an article, as then one can quote exactly the version as it was on that day, even if it may change in the future

OK. Now, I understand better.

**Comment by eu-genia** [comment]
I have replaced "Version of" with "Last edit"

So now we have

Word "stub" = `Stub`
One change element = `Draft `
Anything else = `Last edit XXXX`

Objections?

**Comment by eu-genia** [comment]
Actually the guidelines did specify the old rules
https://betamasaheft.eu/Guidelines/?id=general has

> When you are done with a file, please, add a `<change>`[↗](https://betamasaheft.eu/Guidelines/?id=change) with content **completed** (see [revisions](https://betamasaheft.eu/Guidelines?id=revisions)). If you have reviewed and corrected a file which is now ready and published, please add a change with content reviewed.

see also https://betamasaheft.eu/Guidelines/?id=revisions

> Cataloguer will enter a `<change>`[↗](https://betamasaheft.eu/Guidelines/?id=change) with content **Submitted for review**, once she is finished.
> 
> Reviewer will add a `<change>`[↗](https://betamasaheft.eu/Guidelines/?id=change) with content **Reviewed and Corrected**,
> 
> General editor will enter a `<change>`[↗](https://betamasaheft.eu/Guidelines/?id=change) with value **Approved**


I will now edit the guidelines to reflect the new practice and hope that we will all remember to check them in case of doubt
